lib/src/screens/courses/specificSectionView.dart

```diff
@@ -184,20 +206,46 @@ class _SectionViewState extends State<SectionView> {
   @override
   Widget build(BuildContext context) {
     return PlatformScaffold(
-      body: SliverView(
-        title: "${widget.courseItem.subjectID} ${widget.courseItem.courseID}",
-        children: _getDetails(widget.sectionItem),
-        titleStyle: TextStyle(
-          color: Colors.white,
-        ),
-        actions: [
-          IconButton(
-            onPressed: () {},
-            icon: Icon(PlatformIcons(context).addCircledOutline),
-            tooltip: "Add course to library.",
-          ),
-        ],
-        leading: null,
+      body: FutureBuilder(
+        future: Future.wait([section, course]),
+        initialData: null,
+        builder: (context, AsyncSnapshot<List<dynamic>?> snapshot) {
+          List<dynamic>? data;
+          if (snapshot.hasData) data = snapshot.data;
+
+          if (snapshot.connectionState == ConnectionState.waiting ||
+              data == null) {
+            return Center(
+              child: SpinKitRing(
+                color: AppColors.urbanaOrange,
+              ),
+            );
+          } else if (snapshot.hasError) {
+            final error = snapshot.error;
+            return Center(
+              child: Text(error.toString()),
+            );
```

Comment:
```suggestion
            return Center(
              child: AlertBox(
                child: Text(
                  error.toString(),
                  style: TextStyle(color: Colors.white),
                ),
              ),
            );
```
Perhaps use the `AlertBox` class for any errors?